platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformStateAccessor.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.platform.consensus.ConsensusSnapshot;
+import com.swirlds.platform.event.AncientMode;
+import com.swirlds.platform.system.SoftwareVersion;
+import com.swirlds.platform.system.address.AddressBook;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+
+/**
+ * This interface represents the platform state and provide access to the state's properties.
+ */
+public interface PlatformStateAccessor {
```

Comment:
In addition to my other comment above, it would be best not to have this interface (I didn't realize you're introducing it in this PR, I thought it was an existing one), and instead always use the States API approach everywhere.

Reply:
That would complicate things by a lot with no particular gains. Also, we'd have to support all the "service" infrastructure for the demo apps, which is excessive. We do need two implementation of `PlatformStateAccessor` - one is backed by the platform state (self-sufficient, easy to create and maintain) and another backed by State API (much more flexible, aligned with the other services, but more difficult to maintain).